confd/pkg/backends/calico/client.go

```diff
@@ -121,18 +121,28 @@ func NewCalicoClient(confdConfig *config.Config) (*client, error) {
 	// Create the client.  Initialize the cache revision to 1 so that the watcher
 	// code can handle the first iteration by always rendering.
 	c := &client{
-		client:            bc,
-		cache:             make(map[string]string),
-		peeringCache:      make(map[string]string),
-		cacheRevision:     1,
-		revisionsByPrefix: make(map[string]uint64),
-		nodeMeshEnabled:   nodeMeshEnabled,
-		nodeLabelManager:  newNodeLabelManager(),
-		bgpPeers:          make(map[string]*apiv3.BGPPeer),
-		sourceReady:       make(map[string]bool),
-		nodeListenPorts:   make(map[string]uint16),
-		globalBGPConfig:   cfg,
-		nodeIPs:           make(map[string]struct{}),
+		client:                  bc,
+		cache:                   make(map[string]string),
+		peeringCache:            make(map[string]string),
+		cacheRevision:           1,
+		revisionsByPrefix:       make(map[string]uint64),
+		nodeMeshEnabled:         nodeMeshEnabled,
+		nodeLabelManager:        newNodeLabelManager(),
+		bgpPeers:                make(map[string]*apiv3.BGPPeer),
+		sourceReady:             make(map[string]bool),
+		nodeListenPorts:         make(map[string]uint16),
+		globalBGPConfig:         cfg,
+		nodeIPs:                 make(map[string]struct{}),
+		programmedRouteRefCount: make(map[string]int),
+
+		// Track which routes we have sent, and which we have not. We need maps for
+		// each of the three types of routes we track.
+		programmedRejectRoutesExt: make(map[string]bool),
```

Comment:
This is a bit of a hack, but I think what we need to do in order to track the routes we've sent more precisely. 

Would be better to encap this within a sub-component, which I'll do in a cleanup commit soon.